retail/interactive-tutorial/RetailSearch.Samples/SearchWithPageSizeSample.cs

```diff
@@ -96,6 +96,11 @@ public IEnumerable<SearchResponse> Search(string projectId)
                 Console.WriteLine(item);
                 Console.WriteLine();
             }
+
+            // Paste call with next page token here:
+
+            // Paste call with offset here:
```

Comment:
As discussed before, this will be very confusing for users looking at this code in documentation directly, as this only makes some sense if users are following the tutorial.

Please add the code here that demonstrates your use case.

Also, take a look at https://cloud.google.com/dotnet/docs/reference/help/page-streaming, which documents all the different ways you can use the .NET libraries for paginating results. Decide which use case you are attempting to show with this sample and use similar code to what we have there.

Reply:
Also, in line 79, I'm pretty sure `firstPage` will never be null, it will be an empty IEnumerable.

Reply:
I'm talked with Yana about it. So we need this comments in sample, because these interactive samples should be used step by step.

Reply:
```
string nextPageToken = firstPage.NextPageToken;
itemCount = 0;
SearchRequest searchRequestNextPage = GetSearchRequest("Hoodie", pageSize, projectId, 0, nextPageToken);
IEnumerable<SearchResponse> searchResponseNextPage = client.Search(searchRequestNextPage).AsRawResponses();
SearchResponse nextPage = searchResponseNextPage.FirstOrDefault();

Console.WriteLine("Next page search results:");
foreach (SearchResponse.Types.SearchResult item in nextPage)
{
    itemCount++;
    Console.WriteLine($"Item {itemCount}: ");
    Console.WriteLine(item);
    Console.WriteLine();
}
```

```
int offset = 3;
itemCount = 0;
SearchRequest searchRequestWithOffset = GetSearchRequest("Hoodie", pageSize, projectId, offset);
IEnumerable<SearchResponse> searchResponseWithOffset = client.Search(searchRequestWithOffset).AsRawResponses();
SearchResponse searchWithOffset = searchResponseWithOffset.FirstOrDefault();

Console.WriteLine("Search with offset results:");
foreach (SearchResponse.Types.SearchResult item in searchWithOffset)
{
    itemCount++;
    Console.WriteLine($"Item {itemCount}: ");
    Console.WriteLine(item);
    Console.WriteLine();
}
```

We have realy big code parts and we can't just comment out them in the sample.

Reply:
> Also, in line 79, I'm pretty sure `firstPage` will never be null, it will be an empty IEnumerable.

You're right, I've fixed it.

Reply:
> We have realy big code parts and we can't just comment out them in the sample.

That's part of my point, you really don't need that much code. I believe the use case you are trying to show is the one [described here](https://cloud.google.com/dotnet/docs/reference/help/page-streaming#display-the-next-fixed-sized-page-of-results). Note that when you use AsRawResponse the [page size is not guaranteed](https://cloud.google.com/dotnet/docs/reference/help/page-streaming#readpage). Ideally you would reconsider this code for both the tutorials and documentation, maybe @chenlei1216 can confirm which use case you are trying to show?
But if not, I think that for documentation you can include something as follows, which would make it clear for people not following the tutorial:

```suggestion
            // You can modify the original request to include the next page token and call the service:
            // searchRequest.PageToken = firstPage.NextPageToken;
            // SearchResponse secondPage = client.Search(searchRequest).AsRawResponses().First();
            // You can now iterate over secondPage the same as you did over firstPage.

            // You can also modify the request to include an offset:
            // searchRequest.Offset = 3;
            // SearchResponse firstPageThreeSkipped = client.Search(searchRequest).AsRawResponses().First();
            // You can now iterate over firstPageThreeSkipped the same as you did over firstPage.
```

Reply:
These samples shouldn't be in the documentation. We use them only during interactive tutorials

Reply:
> These samples shouldn't be in the documentation. They are used only for interactive tutorials

They have region tags, they can be added to documentation any time.

Reply:
> > These samples shouldn't be in the documentation. They are used only for interactive tutorials
> 
> They have region tags, they can be added to documentation any time.

Amanda hi!
they have separate code samples for documentation. Ours won't bee used for that.
About having commented code: it can affect user experience, we have step by step instruction in the tutorials to show all features and all these steps with code pieces are included there
If you don't mind, we won't to leave it this way to be aligned wit other tutorials and code samples. 
We have the same approach for all of them, not only for pagination

Reply:
I've sent an internal email, thanks!

Reply:
Region tags have been removed. Could we leave the same comments after that?